app/move/controllers/cancel.js

```diff
@@ -39,12 +39,21 @@ class CancelController extends FormWizardController {
         Object.keys(req.form.options.allFields)
       )
 
-      data.cancellation_reason_comment = data.cancellation_reason_other_comment
+      if (
```

Comment:
I find if/else if blocks a little harder to read and understand. 

I wonder if a switch statement might be easier to read in this instance:

```javascript
const reason = data.cancellation_reason
let comment

switch (reason) {
  case 'cancelled_by_pmu':
    comment = data.cancellation_reason_cancelled_by_pmu_comment
    break
  case 'other':
    comment = data.cancellation_reason_other_comment
    break
  default:
    comment = data.cancellation_reason_comment
    break
}

await moveService.cancel(moveId, {
  reason,
  comment,
})
```

Reply:
We only have a comment for 2 of the reasons, for the other 2 cases, we ignore that option.

The API has the field as optional, so we should only be sending it if we have it.

Would a switch statement without a default option be an acceptable substitute?

Reply:
>We only have a comment for 2 of the reasons, for the other 2 cases, we ignore that option.

Ah, gotcha. Then in that case I think this is a good suggestion:

Would a switch statement without a default option be an acceptable substitute?

Reply:
Personally I prefer object literals and hate the switches

```js
const commentType = {
  cancelled_by_pmu: 'cancellation_reason_cancelled_by_pmu_comment',
  other: 'cancellation_reason_other_comment',
  default: 'cancellation_reason_comment'
}

const reason = data.cancellation_reason
let comment = data[commentType[reason || 'default']]

await moveService.cancel(moveId, {
  reason,
  comment,
})
```

Reply:
@solidgoldpig In this scenario, we don't have a default option for the comment, so there'd be a few more undefined checks that would need to be done. Given that, I don't know if it would be cleaner than the switch statement I've implemented.

Reply:
I'm pretty sure that what I wrote is absolutely equivalent to the switch.

It's a bit of a running gag - any time someone uses a switch statement, I show that you could just as well do it with an object literal. The ghost of Douglas Crockford's "The Good Parts" still stalks this land...

Reply:
@solidgoldpig Ah, sorry - I was unclear. What I meant to say is that you've re-written @teneightfive 's code, which is great - but neither of them correctly solve the problem, as there is no default comment :)

Reply:
sorry, here's the original rewritten then

```js
      const commentType = {
        cancelled_by_pmu: 'cancellation_reason_cancelled_by_pmu_comment',
        other: 'cancellation_reason_other_comment',
      }

      const reason = data.cancellation_reason
      const comment = data[commentType[data.cancellation_reason]]

      await moveService.cancel(moveId, {
        reason,
        ...(comment && {
          comment: comment,
        }),
      })
```

Even simpler!

Reply:
or indeed

```js
      const reason = data.cancellation_reason
      const comment =
        data[`cancellation_reason_${data.cancellation_reason}_comment`]

      await moveService.cancel(moveId, {
        reason,
        ...(comment && {
          comment: comment,
        }),
      })
```

if we're sticking rigorously to that naming convention and don't need the map ;)